src/index.js

```diff
@@ -620,7 +624,12 @@ export class RaspiIOCore extends EventEmitter {
       this.pinMode(pin, SERVO_MODE);
     }
     const period = 1000000 / pinInstance.peripheral.frequency; // in us
-    const pulseWidth = (pinInstance.min + (value / 180) * (pinInstance.max - pinInstance.min)); // in us
+    var pulseWidth;
+    if (value < 544) {
+      pulseWidth = pinInstance.min + constrain(value, 0, 180) / 180 * (pinInstance.max - pinInstance.min);
+    } else {
+      pulseWidth = constrain(value, pinInstance.min, pinInstance.max);
```

Comment:
I'm not exactly following what's going on here, can you explain it more? This feels overly simple to me, but it could just be me not understanding...do we know for certain that this code will ensure that `pulseWidth` will always be less than or equal to `period`?

My understanding was that `min` and `max` were boundary limits on an integer value to be passed in, but that there was no correlation between `min`/`max` and the actual frequency of the PWM, correct? If so, we'll need to do some math to convert from this "abstract" integer to be relative to frequency, like we do when the value is less than `544`.

Also of note: the way this current code works, any value greater than `180` but less than `544` is constrained to `180`, is this desired?

Reply:
> This feels overly simple to me

Music to my ears

> do we know for certain that this code will ensure that pulseWidth will always be less than or equal to period?

No. This assumes that pinInstance.min and pinInstance.max are set correctly for 50hz. We could add this guard, but it doesn't exist in any other J5 plug-in (I think... need to verify).

> My understanding was that min and max were boundary limits on an integer value to be passed in

They are the boundary limits for the pulse width in micro-seconds that can be written to the pin instance. Frequency is always assumed to be 50hz which, unlike pulseWidth range, is very much a standard across all platforms and servos.

> Also of note: the way this current code works, any value greater than 180 but less than 544 is constrained to 180, is this desired?

Yes. If you have seen instances of users passing in values greater than 180° I assume they were doing this as a hack to get around the incorrect pulse width range (1000-2000 will only give you about 60% of a typical servo's range).

Reply:
Oh wait, I ran through some examples and gained a better understanding of this code...and it actually _is_ that simple. After the complications of getting degrees to be converted in conjunction with `min` and `max`, I was thinking "there's no way it's this simple, we need extra conversion logic" but it actually is this simple :D